### PR TITLE
PM User 7 days before and On Expired

### PR DESCRIPTION
**Pull request by yswei93**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -280,9 +280,40 @@ async def clear_expired():
...
+                await member.send("Your VIP experience will expire in 7 days.")
+                logger.info(f"Sent VIP expiration notice to {expiring_user_id}")
+            except Exception as e:
+                logger.error(f"Failed to send expiration message to {expiring_user_id}: {e}")
+
+
 # Main Program Run here
 schedule.every().day.at('00:00').do(lambda: asyncio.create_task(clear_expired()))
-
+schedule.every().day.at('08:00').do(lambda: asyncio.create_task(notify_expiring_expired_vips()))
```
Comment (kingsleylow0327) [inline app.py]: change it to 00:00

File: sql_con.py [hunk 2]
```diff
@@ -273,32 +277,56 @@ def update_expiry_date_by_discord_id(self, discord_id, expiry_date):
...
+    def fetch_expired_vips(self):
+        sql = f"""
+        SELECT discord_id
+        FROM {self.config.TRADE_VOLUME_TABLE}
+        WHERE DATE(vip_expired_date) = CURDATE()
+        """
+        result = self.dbcon_manager(sql, get_all=True)
+        return [row['discord_id'] for row in result] if result else []
+
+    def fetch_seven_days_expiry_vips(self):
```
Comment (kingsleylow0327) [inline sql_con.py]: We can combine this and above function. Since they do the same thing and only differ by number of days, we can make day as a param

**Review by kingsleylow0327 — COMMENTED** [review]
Please take a look at my commend

File: sql_con.py (hunk 2 above)
Comment (yswei93) [reply sql_con.py]: fixed

File: app.py (hunk 1 above)
Comment (yswei93) [reply app.py]: fixed

**Review by kingsleylow0327 — APPROVED** [review]
LGTM

**Pull request merged**